.gitignore

```diff
@@ -27,7 +27,6 @@ envfile
 # kubeconfigs
 kind.kubeconfig
 minikube.kubeconfig
-kubeconfig
```

Comment:
why is this removed?

Reply:
After doing a further audit, I'm confident that this PR updates the entire set of tooling that we provide so that the vanilla `./kubeconfig` will no longer be persisted.

But, because we've been using that generic name for a long time, existing developers will likely still have those files around and so we should keep this entry in place to prevent accidentally including it in their next PR. :(

Also I checked and `make clean` (via `make clean-temporary` still includes a step to explicitly remove `./kubeconfig`, so eventually these will go away from folks' local cloned repos. But it doesn't hurt to keep this in `.gitignore` indefinitely.